core/dbt/task/compare.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import print_function
+
+from dbt.adapters.factory import get_adapter
+from dbt.logger import GLOBAL_LOGGER as logger
+from dbt.runner import RunManager
+from dbt.node_types import NodeType
+from dbt.node_runners import ModelRunner
+from dbt.utils import is_enabled as check_is_enabled
+
+import dbt.ui.printer
+from dbt.task.base_task import BaseTask
+
+
+class CompareTask(BaseTask):
+    def _get_manifest(self):
+        compiler = dbt.compilation.Compiler(self.config)
+        compiler.initialize()
+
+        all_projects = compiler.get_all_projects()
+
+        manifest = dbt.loader.GraphLoader.load_all(self.config, all_projects)
+        return manifest
+
+    def run(self):
+
+        logger.info(
+            "dbt compare: Comparing local codebase to catalog in data warehouse."
+        )
+
+        manifest = self._get_manifest()
+
+        # Look up all of the relations dbt knows about
+        model_relations = set()
+        for node in manifest.nodes.values():
+            node = node.to_dict()
+            is_refable = node["resource_type"] in NodeType.refable()
+            is_enabled = check_is_enabled(node)
+            is_ephemeral = node["config"]["materialized"] == "ephemeral"
+            if is_refable and is_enabled and not is_ephemeral:
+                rel = (node["schema"].lower(), node["alias"].lower())
+                model_relations.add(rel)
+
+        # Look up all of the relations in the DB
+        adapter = get_adapter(self.config)
+        results = adapter.get_catalog(manifest)
+
+        results = [dict(zip(results.column_names, row)) for row in results]
+
+        database_relations = set()
+        for row in results:
+            rel = (row["table_schema"].lower(), row["table_name"].lower())
+            database_relations.add(rel)
+
+        checked_schemas = set([x[0] for x in database_relations])
+
+        logger.info("-" * 40)
```

Comment:
I think this is a little heavy on the dashes. Example output looks like:

```
dbt compare: Comparing local codebase to catalog in data warehouse.
----------------------------------------
dbt compare reviewed the following schemas:
dbt_dbanin
----------------------------------------
----------------------------------------
Warning: We found some discrepancies between the database catalog and the dbt models:
This may indicate that you have outdated relations in your warehouse that can be removed.
----------------------------------------
DIFFERENCE: dbt_dbanin.events
DIFFERENCE: dbt_dbanin.fullevents_showsheet
DIFFERENCE: dbt_dbanin.fullevents_login
DIFFERENCE: dbt_dbanin.fullevents_view
```

We don't have any hard and fast rules about dbt's command line UX, but I do like to keep things mostly consistent when possible. Can you reswizzle this to look something like:

```
Comparing local models to the database catalog. Checking schemas:
- dbt_dbanin
- dbt_dbanin_abc
- dbt_dbanin_def

Warning: The following relations do not match any models found in this project:
table "dbt_dbanin"."events"
view "dbt_dbanin"."fullevents_show"
table "dbt_dbanin"."fullevents_login"
table "dbt_dbanin"."fullevents_view"
```

There are a couple of subtle changes here that i think are important:
1) Use a formal 3rd person tense rather than we/you. I'm not actually opposed to the style you used here, but it's not something we do elsewhere in dbt AFAIK, and I think it's important to be consistent
2) Remove the `DIFFERENCE` from each problem line. This will make it easy to copy/paste these things into an editor where you can drop them.
3) Add the relation type back in. Sorry for leading you astray earlier -- i didn't realize this is where you were using the relation type, and I think it's a super good idea. I like that you can prepend all of these with `drop` and get some runnable queries (if you so choose)
4) Use quoted relation names (more on that below)

If you want to get cute, I think you can use `dbt.ui.printer.green()` and `dbt.ui.printer.yellow()` for the status messages :)